packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx

```diff
@@ -151,109 +227,205 @@ export const LivenessCameraModule = (
-                send({ type: 'CANCEL' });
-              }}
-            >
-              {renderErrorModal({
-                errorState,
-                overrideErrorDisplayText: errorDisplayText,
-              })}
-            </ErrorView>
+          <video
+            ref={videoRef}
+            muted
+            autoPlay
+            playsInline
+            style={{ transform: 'scaleX(-1)' }}
+            width={mediaWidth}
+            height={mediaHeight}
+            onCanPlay={handleMediaPlay}
+            data-testid="video"
+            className={LivenessClassNames.Video}
+          />
+          <Flex
+            className={classNames(
+              LivenessClassNames.OvalCanvas,
+              isRecordingOnMobile && `${LivenessClassNames.OvalCanvas}--mobile`,
+              isRecordingStopped && LivenessClassNames.FadeOut
+            )}
+          >
+            <View as="canvas" ref={canvasRef} />
+          </Flex>
+
+          {isRecording && (
+            <RecordingIcon recordingIndicatorText={recordingIndicatorText} />
+          )}
+
+          {!isStartView && !isWaitingForCamera && !isCheckSucceeded && (
+            <CancelButton cancelLivenessCheckText={cancelLivenessCheckText} />
           )}
 
-          {/* 
+          <Overlay
+            anchorOrigin={{
+              horizontal: 'center',
+              vertical:
+                isRecording && !isFlashingFreshness ? 'start' : 'space-between',
+            }}
+            className={LivenessClassNames.InstructionOverlay}
+          >
+            <Hint hintDisplayText={hintDisplayText} />
+
+            {errorState && (
+              <ErrorView
+                onRetry={() => {
+                  send({ type: 'CANCEL' });
+                }}
+              >
+                {renderErrorModal({
+                  errorState,
+                  overrideErrorDisplayText: errorDisplayText,
+                })}
+              </ErrorView>
+            )}
+
+            {/* 
               We only want to show the MatchIndicator when we're recording
               and when the face is in either the too far state, or the 
               initial face identified state. Using the a memoized MatchIndicator here
               so that even when this component re-renders the indicator is only
               re-rendered if the percentage prop changes.
             */}
-          {isRecording &&
-          !isFlashingFreshness &&
-          showMatchIndicatorStates.includes(faceMatchState!) ? (
-            <MemoizedMatchIndicator
-              percentage={Math.ceil(faceMatchPercentage!)}
-            />
-          ) : null}
-        </Overlay>
-      </View>
-    </Flex>
+            {isRecording &&
+            !isFlashingFreshness &&
+            showMatchIndicatorStates.includes(faceMatchState!) ? (
+              <MemoizedMatchIndicator
+                percentage={Math.ceil(faceMatchPercentage!)}
+              />
+            ) : null}
```

Comment:
Concerned the conditional logic going on here is fragile and well be difficult to maintain.

Only calling out here but imo applies to the rest of the file, there is a lot of opportunity to cross cut concerns by scoping components to specific needs